lib/modem_jwt/modem_jwt.c

```diff
@@ -148,6 +158,43 @@ int modem_jwt_generate(struct jwt_data *const jwt)
 	return ret;
 }
 
+const char *modem_jwt_get_uuid()
+{
+	struct jwt_data jwt = {0};
+
+	int rc = modem_jwt_generate(&jwt);
+	if (rc) {
+		modem_jwt_free(jwt.jwt_buf);
+		return NULL;
+	}
+
+	char *p = strchr(jwt.jwt_buf, '.') + 1;
+	char *end = strrchr(jwt.jwt_buf, '.');
+	char *payload =  k_calloc(1, 256);
+	size_t len;
+
+	rc = base64_decode(payload, 256, &len, p, end - p);
+	if (rc) {
+		k_free(payload);
+		modem_jwt_free(jwt.jwt_buf);
+		return NULL;
+	}
+	const char *iss = "iss\":\"nRF9160.";
```

Comment:
it would be more future-proof if you did not include the device name in this search.